sgt-back-end/index.js

```diff
@@ -0,0 +1,154 @@
+const pg = require('pg');
+const fs = require('fs');
+const express = require('express');
+const app = express();
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+})
+
+app.use(express.json());
+
+app.get('/api/grades', (req, res, next) => {
+  const sql = `
+    select *
+      from "grades"
+  `;
+db.query(sql)
+  .then(result => {
+    res.status(200).json(result.rows);
+  })
+  .catch(err => {
+    console.error(err);
+    res.status(500).json({
+      error: 'An unexpected error occurred.'
+    });
+  })
+});
+
+app.post('/api/grades', (req, res, next) => {
+  const score = Number(req.body.score);
+  if (req.body.name === undefined || req.body.course === undefined || req.body.score === undefined) {
+    res.status(400).json({
+      error: 'missing "name", "course", or "score"'
+    });
+    return;
+  } else if (score <= 0 || score > 100) {
+    res.status(400).json({
+      error: 'score must be a positive integer from 0 to 100'
+    });
+    return;
+  } else if (!Number.isInteger(score)) {
+    res.status(400).json({
+      error: 'score must be a positive integer from 0 to 100'
+    });
+    return;
+  }
+  const sql = `
+    insert into "grades" ("name", "course", "score")
+    values ($1, $2, $3)
+  `;
+  const params = [req.body.name, req.body.course, req.body.score];
+  db.query(sql, params)
+    .then(result => {
+      res.status(201).json(req.body)
```

Comment:
The client should receive the generated id from the database too. If you include a `returning` clause in your insert statement, then you can send back `result.rows[0]` instead of `req.body`.